sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -494,4 +513,270 @@ export class AppConfigurationClient {
   updateSyncToken(syncToken: string): void {
     this._syncTokens.addSyncTokenFromHeaderValue(syncToken);
   }
+
+  /**
+   * Create a snapshot for Azure App Configuration service, failing if it
+   * already exists.
+   *
+   * Example usage:
+   * ```ts
+   * const result = await client.createSnapshot("MySnapshot", { key: "MyKey", label: "MyLabel"});
+   * ```
+   * @param configurationSetting - A configuration setting.
+   * @param options - Optional parameters for the request.
+   */
+  async beginCreateSnapshot(
+    snapshot: SnapshotInfo,
+    options: CreateSnapshotOptions = {}
+  ): Promise<SimplePollerLike<OperationState<CreateSnapshotResponse>, CreateSnapshotResponse>> {
+    logger.info("[createSnapshot] Creating snapshot");
+    const self = this;
+    let statusOfThePoller:string;
+    function createLro(): LongRunningOperation<unknown> {
+      return {
+        async sendInitialRequest(): Promise<LroResponse<CreateSnapshotResponse>> {
+          return tracingClient.withSpan(
+            `${AppConfigurationClient.name}.createSnapshot`,
+            options,
+            async (updatedOptions) => {
+              const originalResponse = await self.client.createSnapshot(
+                snapshot.name,
+                { ...snapshot },
+                {
+                  ...updatedOptions,
+                }
+              );
+              const res = transformSnapshotResponse(originalResponse)
+              return {
+                flatResponse: res,
+                rawResponse: {
+                  statusCode: res._response.status,
+                  headers: res._response.headers.toJson(),
+                  body: res._response.parsedBody,
+                },
+              };
+            }
+          );
+        },
+        async sendPollRequest(): Promise<LroResponse<unknown>> {
+          return tracingClient.withSpan(
+            `${AppConfigurationClient.name}.getOperationDetails`,
+            options,
+            async (updatedOptions) => {
+              if (statusOfThePoller === "Succeeded") {
+                const response = await self.getSnapshot(snapshot.name, updatedOptions);
```

Comment:
Feedback for the poller: 
Consider adding a new param that takes a new method that will be called once the polling is succeeded rather than hacking our way around the `sendPollRequest` for the last response to be returned to the user.